src/Catalog/Helpers/FeedHelpers.cs

```
@@ -355,6 +308,78 @@ private static DateTime ForceUtc(DateTime date)
+            HttpResponseMessage response = null;
+            try
+            {
+                using (telemetryService.TrackDuration(
+                    TelemetryConstants.PackageDownloadSeconds,
+                    new Dictionary<string, string>()
+                    {
+                        { TelemetryConstants.Id, feedPackageDetails.PackageId?.ToLowerInvariant() },
+                        { TelemetryConstants.Version, feedPackageDetails.PackageVersion?.ToLowerInvariant() },
+                    }))
+                {
+                    response = await client.GetAsync(packageUri, cancellationToken);
+                }
+            }
+            catch (TaskCanceledException tce)
+            {
+                // If the HTTP request timed out, a TaskCanceledException will be thrown.
+                throw new HttpClientTimeoutException($"HttpClient request timed out in {nameof(FeedHelpers.DownloadMetadata2CatalogAsync)}.", tce);
+            }
+
+            if (response.IsSuccessStatusCode)
+            {
+                using (var stream = await response.Content.ReadAsStreamAsync())
+                {
+                    workItem.PackageCatalogItem = Utils.CreateCatalogItem(
+                        feedPackageDetails.ContentUri.ToString(),
+                        stream,
+                        feedPackageDetails.CreatedDate,
+                        feedPackageDetails.LastEditedDate,
+                        feedPackageDetails.PublishedDate);
+
+                    if (workItem.PackageCatalogItem != null)
+                    {
+                        logger?.LogInformation("Add metadata from: {PackageDetailsContentUri}", feedPackageDetails.ContentUri);
+                    }
+                    else
+                    {
+                        logger?.LogWarning("Unable to extract metadata from: {PackageDetailsContentUri}", feedPackageDetails.ContentUri);
+                    }
```

Is this the right behavior? This silent failure means that we have a package on v2 but not v3.

Also, could you change this line to log the full exception instead of just its type and message: [https://github.com/NuGet/NuGet.Services.Metadata/blob/master/src/Catalog/Helpers/Utils.cs#L513](https://github.com/NuGet/NuGet.Services.Metadata/blob/master/src/Catalog/Helpers/Utils.cs#L513)

---

Never mind this comment as well.